src/app/betting/feature/accept-bets-page.component.ts

```diff
@@ -0,0 +1,176 @@
+                </div>
+                <div>
+                  {{ userNameAndScore() }} wagers:
+                  {{ bet.yourWager }}
+                </div>
+                <div>{{ bet.description }}</div>
+              </div>
+            </td>
+            <!-- Accept/Reject Buttons -->
+            <td>
+              <div
+                class="text-right flex gap-2 flex-column md:flex-row justify-content-end"
+              >
+                <p-button
+                  label="Accept Bet"
+                  icon="pi pi-check"
+                  styleClass="w-full"
+                  (onClick)="acceptBet(bet.id)"
+                  [disabled]="
+                    bet.yourWager > (userPlayer()?.score ?? 0) ||
+                    bet.requesterWager > (bet.requesterScore ?? 0)
+                  "
+                  [loading]="submitting()"
+                />
+                <p-button
+                  label="Reject Bet"
+                  icon="pi pi-times"
+                  styleClass="w-full"
+                  (onClick)="rejectBet(bet.id)"
+                  [loading]="submitting()"
+                />
+              </div>
+            </td>
+          </tr>
+        </ng-template>
+      </p-table>
+    }
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export default class PlaceBetChoosePlayerPageComponent {
+  private readonly playerService = inject(PlayerService);
+  private readonly betService = inject(BetService);
+  private readonly messageService = inject(MessageService);
+
+  readonly userPlayer = this.playerService.userPlayer;
+  readonly bets = this.betService.bets;
+
+  readonly submitting = signal(false);
+
+  private readonly betsAwaitingUser = computed(() => {
+    return this.bets()?.filter(
+      (bet) =>
+        bet.opponent_player_id === this.userPlayer()?.player_id &&
+        bet.status === BetStatus.PendingAcceptance,
+    );
+  });
+
+  readonly displayBets = computed(() =>
+    this.betsAwaitingUser()?.map((bet) => {
+      const players = this.playerService.players();
+      const requester = players?.filter(
+        (player) => player.player_id === bet.requester_player_id,
+      )[0];
+      return {
+        requesterName:
+          (requester?.display_name ?? 'Requester') +
+          ' (' +
+          requester?.score +
+          ' points)',
+        requesterScore: requester?.score,
+        requesterWager: bet.requester_wager,
+        yourWager: bet.opponent_wager,
+        description: bet.description,
+        id: bet.id,
+      };
+    }),
+  );
+
+  readonly userNameAndScore = computed(() => {
+    const userPlayer = this.userPlayer();
+    return (
+      (userPlayer?.display_name ?? 'Player') +
+      ' (' +
+      userPlayer?.score +
+      ' points)'
+    );
+  });
+
+  async acceptBet(id: number) {
+    this.submitting.set(true);
+    await showMessageOnError(
+      this.betService.acceptBet(id),
+      this.messageService,
+    );
+    this.submitting.set(false);
```

Comment:
I would use `confirmBackendAction` for both of these. If the user drunkenly taps the wrong button, we wouldn't want it to immediately submit.

Side note: I've started doing this thing recently where I'll make the type of an ID `Model['id']`. It doesn't make a huge difference, since it's just referencing the normal type, but I like it. Especially in case the type of the `id` were to at some point change from `number` to `string`. Feel free to use or not.

```typescript
async acceptBet(betId: BetModel['id']): Promise<void> {
  confirmBackendAction({ 
    action: async () => this.betService.acceptBet(betId),
    confirmationMessageText: 'Are you sure you want to accept this bet?',
    successMessageText: 'Bet accepted',
    submittingSignal: this.submitting,
    confirmationService: this.confirmationService,
    messageService: this.messageService,
    successNavigation: null,
  });
}
```

Reply:
I had tried this originally and abandoned it because my submittingSignal wasn't getting reset. I had been trying to pass successNavigation: ".", not realizing I could just pass null.